### ISSUE TITLE
[Feature Request] add loader to use other loader

### ISSUE THREAD

**Issue by dimdamin** [aminought/firefox-second-sidebar#5]
Hi, can You add loader to use other loader, i use [**Xiao Fleaud**](https://github.com/xiaoxiaoflood/firefox-scripts)
my load script
````js
// ==UserScript==
// @name            Second Sidebar for Firefox
// @description     A Firefox userChrome.js script for adding a second sidebar with web panels like in Vivaldi/Floorp/Zen.
// @author          aminought
// @include         main
// @homepageURL     https://github.com/aminought/firefox-second-sidebar
// ==/UserScript==

if (location.href.startsWith("chrome://browser/content/browser.x")) {
	(async url => {
		(await ChromeUtils.compileScript(`data:,"use strict";import("${url}").catch(console.error)`)).executeInGlobal(window);
	})(Services.io.newURI(Components.stack.filename).resolve("second_sidebar.uc.mjs"));
}
````

**Comment by aminought** [comment]
I'm sorry, I'm not really a web developer, and I don't understand some things. What exactly is going on in this code, and why doesn't it work without it?

**Comment by aminought** [comment]
I have done some research on the topic and tried the loader myself. To be honest, I wasn't very impressed, but that doesn't matter. I just can't see this script fitting into the codebase at the moment. However, I can include it in the installation instructions and mention you there.

**Comment by dimdamin** [comment]
Thank You. Yes you can...

**Comment by benzBrake** [comment]
I'm using [alice0775's loader](https://github.com/alice0775/userChrome.js).
In addition to this script, I also need to modify `second_sidebar.uc.mjs`
change
```
UC_API.Runtime.startupFinished().then(run);
```
to
```
run();
```

**Comment by aminought** [comment]
@benzBrake does this loader support `*.mjs` scripts?

**Comment by benzBrake** [comment]
> @benzBrake does this loader support `*.mjs` scripts?


As far as I know, only fx-autoconfig adds the function of automatically loading mjs, and other Loaders require the script from @dimdamin

**Comment by aminought** [comment]
> In addition to this script, I also need to modify `second_sidebar.uc.mjs`

I think I'll replace it with something like this:
```js
if (typeof UC_API !== 'undefined') {
  UC_API.Runtime.startupFinished().then(run);
} else {
  run();
}
```

**Comment by dimdamin** [comment]
FYI
```
// ==UserScript==
// @name            Second Sidebar for Firefox
// @description     A Firefox userChrome.js script for adding a second sidebar with web panels like in Vivaldi/Floorp/Zen.
// @author          aminought
// @homepageURL     https://github.com/aminought/firefox-second-sidebar
// ==/UserScript==

import { SidebarDecorator } from "./second_sidebar/sidebar_decorator.mjs";
import { SidebarInjector } from "./second_sidebar/sidebar_injector.mjs";

//const run = () => {
(async () => {
  await delayedStartupPromise;
  SidebarInjector.inject();
  SidebarDecorator.decorate();
})();

//UC_API.Runtime.startupFinished().then(run);
```

**Comment by aminought** [comment]
> await delayedStartupPromise;

[It's not that simple](https://github.com/MrOtherGuy/fx-autoconfig/blob/fe783f2c72388f64fd7ea0ee67617c6fd32f2261/profile/chrome/utils/utils.sys.mjs#L757).

**Comment by aminought** [comment]
OK:
```js
if (typeof UC_API !== 'undefined') {
  UC_API.Runtime.startupFinished().then(run);
} else {
  delayedStartupPromise.then(run);
}
```

**Comment by dimdamin** [comment]
Firefox native function => work on all loader`szzzzz
```
(async () => {
  await delayedStartupPromise;
  SidebarInjector.inject();
  SidebarDecorator.decorate();
})();
```
native function work on all loader`szzzzz
```
delayedStartupPromise.then(run);
```

external function work only in fx-autoconfig
```
UC_API.Runtime.startupFinished().then(run);
```
please close this  issues with don`t planet...

**Comment by aminought** [comment]
> OK:
> 
> ```js
> if (typeof UC_API !== 'undefined') {
>   UC_API.Runtime.startupFinished().then(run);
> } else {
>   delayedStartupPromise.then(run);
> }
> ```

OMG, this code works on all loaders.

**Comment by aminought** [comment]
Let's figure out something. This is a script for `fx-autoconfig`, which I'm only testing in this loader. However, I have tried to write the rest of the code, excluding `second_sidebar.uc.mjs`, so that it will be supported everywhere. For `second_sidebar.uc.mjs', I have proposed a solution that will work everywhere, but will take advantage of `fx-autoconfig' when used.

@dimdamin I don't enjoy your way of communication. I'm starting to regret not using the `fx-autoconfig` API as much as possible.
